src/Spectre.Data/Datasets/BasicTextDataset.cs

```diff
@@ -0,0 +1,307 @@
+        {
+            get { return _metadata; }
+            private set { _metadata = value; }
+        }
+        /// <summary>
+        /// See <see cref="IDataset"/> for description.
+        /// </summary>
+        public IEnumerable<SpatialCoordinates> SpacialCoordinates
+        {
+            get { return _spatialCoordinates; }
+            private set { _spatialCoordinates = value as List<SpatialCoordinates>; }
+        }
+        /// <summary>
+        /// Method for creating new dataset from text file, overwriting current data.
+        /// </summary>
+        /// <param name="filePath">Path to a text file.</param>
+        /// <exception cref="Exception">Thrown where there is a problem with file loading.</exception>
+        public void CreateFromFile(string filePath)
+        {
+            try
+            {
+                StreamReader sr = new StreamReader(filePath);
+                var metadata = sr.ReadLine(); // global metadata
+                var mzValues = sr.ReadLine()?.Split(null);
+                _mz = new double[mzValues.Length];
+                for (int i = 0; i < _mz.Length; i++)
+                    if (!double.TryParse(mzValues[i], NumberStyles.Any, CultureInfo.InvariantCulture, out _mz[i]))
+                        _mz[i] = double.NaN;
+            }
+            catch (Exception e)
+            {
+                throw new Exception("File " + filePath + " could not be read.", e);
+            }
+
+            _intensity = new List<double[]>();
+            _spatialCoordinates = new List<SpatialCoordinates>();
+            AppendFromFile(filePath);
+        }
+        /// <summary>
+        /// See <see cref="IDataset"/> for description.
+        /// </summary>
+        /// <exception cref="InvalidDataException">Throws when the data is null or 
+        /// the length of data is not matching the dataset length.</exception>
+        public void CreateFromRawData(double[] mz, double[,] data)
+        {
+            if (mz == null || data == null)
+                throw new InvalidDataException("The input data is null.");
+            if (mz.Length != data.GetLength(1))
+                throw new InvalidDataException("Length of the data must be equal to length of m/z values.");
+
+            _intensity = new List<double[]>();
+            _spatialCoordinates = new List<SpatialCoordinates>();
+            _mz = mz;
+            AppendFromRawData(data);
+        }
+
+        /// <summary>
+        /// Method parsing the whole text file and initializing the array
+        /// with found data.
+        /// </summary>
+        /// <param name="filePath">Path to the text file.</param>
+        /// <exception cref="InvalidDataException">Thrown where there is a problem with file parsing.</exception>
+        public void AppendFromFile(string filePath)
+        {
+            //TODO: Specifying formalized format of data in text files.
+            //TODO: Safety check, format check
+            try
```

Comment:
I think the contents could be someday divided into several generic txt parsing methods, so it will look more clear.